massa-execution-worker/src/execution.rs

```diff
@@ -293,6 +296,44 @@ impl ExecutionState {
         HistorySearchResult::NotFound
     }
 
+    /// Lazily query (from end to beginning) the active datastore entry of an address at a given slot.
+    /// Returns None if the datastore entry could not be determined from the active history.
+    pub fn fetch_active_history_data_entry(
+        &self,
+        slot: Slot,
+        addr: &Address,
+        key: &Hash,
+    ) -> HistorySearchResult<Vec<u8>> {
+        self.verify_active_slot(slot);
+
+        if let Some(n) = self.get_active_index(slot) {
+            let iter = self.active_history.iter().rev().skip(n);
```

Comment:
I think the rev and skip has been inverted.
Example : 
We send slot 6 in this context : 
```
Slots: 5 6 7 8 9 10
Index: 0 1 2 3 4 5
              ^

Rev : 10 9 8 7 6 5
Skip : 9 8 7 6 5
```
but I think you want to have : 
```
Skip : 6 7 8 9 10
Rev : 10 9 8 7 6
```
The behavior is already like that in `fetch_active_history_balance`

Reply:
This has been already thought through, see https://github.com/massalabs/massa/pull/2590#discussion_r883771386

Reply:
Yes and on this PR we changed to `self.active_history.iter().skip(n).rev();` here you have : `self.active_history.iter().rev().skip(n);`

Reply:
Done.